gensim/models/wrappers/ldamallet.py

```diff
@@ -244,3 +228,27 @@ def show_topic(self, topicid, topn=10):
 
     def print_topic(self, topicid, topn=10):
         return ' + '.join(['%.3f*%s' % v for v in self.show_topic(topicid, topn)])
+
+
+    def read_doctopics(self, fname, eps=1e-6):
+        """
+        Yield document topic vectors from MALLET's "doc-topics" format, as sparse gensim vectors.
+
+        """
+        fin = utils.smart_open(fname)
+        first_line = next(fin)
+        fin.close()
```

Comment:
Feel iffy about opening and closing like this.

Would this accomplish the same or similar if we moved the check into the loop?

```
with utils.smart_open(fname) as fin:
      for lineno, line in enumerate(fin):
           if lineno == 0 and line.startswith("#doc "): # first line and it is the header line, no need to process it
                continue 
           parts = line.split()[2:]  # skip "doc" and "source" columns
           ...
```